tests/integ/sagemaker/lineage/conftest.py

```diff
@@ -709,27 +721,41 @@ def static_model_artifact(sagemaker_session, static_pipeline_execution_arn):
             )
         )
 
-    yield artifact.ModelArtifact.load(
-        artifacts[0]["ArtifactArn"], sagemaker_session=sagemaker_session
-    )
+    artifact_arn = artifacts[0]["ArtifactArn"]
+    logging.info(f"Using static model artifact {artifact_arn}")
+    yield artifact.ModelArtifact.load(artifact_arn, sagemaker_session=sagemaker_session)
 
 
 @pytest.fixture
 def static_dataset_artifact(static_model_artifact, sagemaker_session):
+    model_artifact_arn = static_model_artifact.artifact_arn
     dataset_associations = sagemaker_session.sagemaker_client.list_associations(
-        DestinationArn=static_model_artifact.artifact_arn, SourceType="DataSet"
+        DestinationArn=model_artifact_arn, SourceType="DataSet"
+    )
+    logging.info(
+        f"Found {len(dataset_associations)} associated with model artifact {model_artifact_arn}"
     )
     if len(dataset_associations["AssociationSummaries"]) == 0:
         # no directly associated dataset. work backwards from the model
         model_associations = sagemaker_session.sagemaker_client.list_associations(
-            DestinationArn=static_model_artifact.artifact_arn, SourceType="Model"
-        )
+            DestinationArn=model_artifact_arn, SourceType="Model"
+        )["AssociationSummaries"]
```

Comment:
Won't this throw an exception if list_associations is empty or null?

Reply:
`list_associations` here is the auto generated boto method for calling `ListAssociations`. as long as the request is successful a response will be returned with a field of AssociationSummaries - which may be empty or not but it will be there resulting in this being successful. so the response being null here is unlikely.